android/src/main/java/com/mattermost/networkclient/APIClientModule.kt

```diff
@@ -56,33 +75,56 @@ class APIClientModule(reactContext: ReactApplicationContext) : ReactContextBaseJ
 
     @ReactMethod
     fun invalidateClientFor(baseUrl: String, promise: Promise) {
+        // Don't trust user input...
+        val url = baseUrl.toHttpUrl().toString();
+
         try {
-            sessionsRequest.remove(baseUrl);
-            sessionsClient.remove(baseUrl);
-            promise.resolve(sessionsRequest.keys);
+            sessionsClient.remove(url);
+            sessionsConfig.remove(url);
+            promise.resolve(sessionsClient.keys);
         } catch (err: Throwable) {
             promise.reject(err)
         }
     }
 
     @ReactMethod
     fun get(baseUrl: String, endpoint: String, options: ReadableMap, promise: Promise) {
+        // Don't trust user input...
+        val url = baseUrl.toHttpUrl().toString();
```

Comment:
Can baseUrl be null ? Like have we tested with 'wrong data' ?

Reply:
I think @migbot has a change coming up soon that wraps .toHttpUrl in a try catch